R/transformAbundance.R

```diff
@@ -0,0 +1,400 @@
+#' object.
+#'
+#' @param abund_values
+#' A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#'   to be transformed.
+#'
+#' @param transform
+#' A single character value for selecting the transformation method.
+#'
+#' @param name
+#' A single character value specifying the name of transformed abundance table.
+#'
+#' @param pseudocount
+#' Boolean or numeric value deciding whether pseudocount is added. Numerical
+#' value specifies the value of pseudocount. When selecting pseudocount==TRUE,
+#' pseudocount is 1.
+#'
+#' @param scale
+#' A numeric value for scaling. Values are multiplied with specified value.
+#'
+#' @param target
+#' A single character value specifying, if z transformation is applied
+#' for samples or features.
+#'
+#' @details
+#' \code{transformAbundance} applies transformation to abundance table.
+#' Transformation methods include
+#'
+#' \itemize{
+#'
+#'     \item{"relabundance" }{Transforms abundances to relative.
+#'     ($\frac{x}{x_{tot}}$, where $x$ is a single value and $x_{tot}$ is the sum of
+#'     all values.)}
+#'
+#'     \item{"Z" }{Z-transform or Z-standardization can be used for normalizing the data.
+#'     It can be done for samples or features by specifying 'target'. However, Z-transform
+#'     done for features can give misleading results. By default, it is done for samples.
+#'     ($\frac{x + µ}{σ}$, where $x$ is a single value, $µ$ is the mean of the sample, and
+#'     $σ$ is the standard deviation of the sample.)}
+#'
+#'     \item{"log10" }{log10 transform can be used for reducing the skewness of the data.
+#'     ($log_{10}x$, where $x$ is a single value of data.)}
+#'
+#'     \item{"log10p }{log10 transform can be used for reducing the skewness of the data.
+#'     With log10p, pseudo count is added before log10 transformation.
+#'     ($log_{10}x$, where $x$ is a single value of data.)}
+#'
+#'     \item{"hellinger" }{Hellinger transform can be used for reducing the impact of
+#'     extreme data points. ($\sqrt{\frac{x}{x_{tot}}}$, where $x$ is a single value and $x_{tot}$ is the sum of
+#'     all values.)}
+#'
+#'     \item{"identity" }{Identity is the data itself without transform.}
+#'
+#'     \item{"clr" }{Centered log ratio (clr) transform can be used for reducing the
+#'     skewness of data and for centering it.
+#'     ($log_{10}x_{r}) - log_{10}µ_{r}$, where $x_{r}$ is a single relative value, $µ_{r}$ is
+#'     the mean of relative values of whole sample.}
+#'
+#' }
+#'
+#' @references
+#' If there is references Add information here......................................
+#'
+#' @return \code{transformAbundance} returns \code{x} with additional, transformed
+#' abundance table named \code{*name*} in the \code{\link{assay}}.
+#' \code{getTransformAbundance} returns transformed abundance table.
+#'
+#' @seealso
+#' If there is something Add information here......................................
```

Comment:
I think we can remove this for now

Reply:
the `relAbundanceCounts` reference should appear here

Reply:
Good point. 

Should we consider whether `relAbundanceCounts` is needed as an independent function, if `transformAbundance` does the job and is more generic? Is there added value from having `relAbundanceCounts`?

Reply:
Yesterday I would have said yes, now I am not so sure. It is probably the most used one I guess, especially for plotting, so thats why I implemented it

I think `relAbundanceCounts` should most likely be a wrapper around `transformAbundance`, but if we think broad picture I could also see a more generic function name `transformCounts` being used. The names aside, I think the implementation should be largely structured the same as `relAbundanceCounts` and `normalizeCounts`, which is already the case, but the redundancies should be removed

Reply:
Yes to all, I think this would be a clear solution.

Reply:
transformAbundance's name changed to transformCounts